File: 1491-Average Salary Excluding the Minimum and Maximum Salary/solution.java
class Solution {
    public double average(int[] salary) {
        final int N = salary.length;
        int min=salary[0],max=salary[0];
        double sum = salary[0];
        for (int i = 1; i < N; i++) {
            if (salary[i]<min) min=salary[i];
            if (salary[i]>max) max=salary[i];
            sum += salary[i];
        }
        sum = sum - max - min;
        return sum / (N - 2);
    }
}

File: 1491-Average Salary Excluding the Minimum and Maximum Salary/solution.py
class Solution:
    def average(self, salary: List[int]) -> float:
        min_salary = min(salary)
        max_salary = max(salary)
        total_salary = sum(salary) - min_salary - max_salary
        num_employees = len(salary) - 2
        return total_salary / num_employees